geofileops/fileops.py

```diff
@@ -1499,7 +1487,7 @@ def _read_file_base_fiona(
     )
 
     # Set the index to the backed-up fid
-    if fid_as_index:
+    if fid_as_index and "__TMP_GEOFILEOPS_FID" in result_gdf.columns:
```

Comment:
I think... not sure, that the problem is that the code below should be moved to right before this line... and then this change can be reverted. The `tmp_fid_path` already is being removed before the file is read, so the `__TMP_...` column (or any column) isn't there? This would also mean that this code hasn't been working for a long time... and the tests aren't picking this up either?

``` python
     finally:
            if ConfigOptions.remove_temp_files and tmp_fid_path.parent.exists():
                shutil.rmtree(tmp_fid_path, ignore_errors=True)
```